UnityProject/Assets/Scripts/Core/Chat/ChatRelay.cs

```diff
@@ -164,6 +164,7 @@ bool DistanceCheck(Vector3 playerPos)
 				foreach (Collider2D coll in npcs)
 				{
 					var npcPosition = coll.gameObject.AssumedWorldPosServer();
+					if ((npcPosition - chatEvent.position).magnitude > 16) continue;
```

Comment:
Vector3.Distance is more accurate because each unit is a single tile; but this works still.